hazelcast/src/main/java/com/hazelcast/spi/discovery/integration/DiscoveryServiceProvider.java

```diff
@@ -25,18 +27,23 @@
  * provide this ability. Every service should have its own provider, however in rare cases a single provider might
  * create different <tt>DiscoveryService</tt> implementations based on the provided {@link DiscoveryMode} or other
  * configuration details.
+ *
+ * @since 3.6
  */
+@Beta
 public interface DiscoveryServiceProvider {
 
     /**
      * Instantiates a new instance of the {@link DiscoveryService}.
      *
-     * @param discoveryMode            the current discovery mode
-     * @param discoveryStrategiesConfig the configuration parsed from the XML or provided using the programmatic API
-     * @param configClassLoader        the classloader set in the configuration
+     * @param member            the current <tt>Member</tt> instance if created on member nodes, otherwise <tt>null</tt>
+     * @param logger            the <tt>ILogger</tt> instance to be used for logging output
+     * @param discoveryMode     the current discovery mode
+     * @param discoveryConfig   the configuration parsed from the XML or provided using the programmatic API
+     * @param configClassLoader the classloader set in the configuration
      * @return a new instance of the discovery service
      */
-    DiscoveryService newDiscoveryService(DiscoveryMode discoveryMode, DiscoveryStrategiesConfig discoveryStrategiesConfig,
-                                         ClassLoader configClassLoader);
+    DiscoveryService newDiscoveryService(Member member, ILogger logger, DiscoveryMode discoveryMode,
```

Comment:
One design trick. Passing arguments sucks because if you add an argument, you break the interface. If you use an argument object, you can easily add new arguments without causing any problems. 

Something like this:
```
public interface DiscoveryServiceProvider {
    DiscoveryService newDiscoveryService(DiscoveryServiceProviderSetting settings);
}

public class DiscoveryServiceProviderSetting{
    public Member member;
    public ILogger logger;
    public DiscoveryMode discoveryMode;
    public DiscoveryConfig discoveryConfig;
    public ClassLoader configClassLoader;
}
```

This gives you a lot more flexibility + I find methods with many arguments awful to read due to lack of named arguments.